File: android/app/src/main/java/ai/diffforge/haider/daemon/NativeDaemon.java
package ai.diffforge.haider.daemon;

import android.content.Context;
import androidx.annotation.Keep;

/** Frozen JNI v1 owner. Loading is explicit and only happens on the native owner thread. */
@Keep
public final class NativeDaemon {
    private NativeDaemon() {}
    public static native String nativeVersion();
    public static native int nativeInit(Context context, String pathsJson);
    public static native int nativeStart(byte[] vaultDek, String policyJson);
    public static native String nativeObserve();
    public static native int nativeShutdown(boolean forced, long deadlineMs);
    public static native void nativeRelease();
}

File: android/spike/host/src/main/java/ai/diffforge/haider/daemon/NativeDaemon.kt
package ai.diffforge.haider.daemon

import android.content.Context

/** Frozen JNI v1 surface exercised by the disposable diagnostic host. */
class NativeDaemon private constructor() {
    companion object {
        init { System.loadLibrary("haider") }
        @JvmStatic external fun nativeVersion(): String
        @JvmStatic external fun nativeInit(context: Context, pathsJson: String): Int
        @JvmStatic external fun nativeStart(vaultDek: ByteArray, policyJson: String): Int
        @JvmStatic external fun nativeObserve(): String
        @JvmStatic external fun nativeShutdown(forced: Boolean, deadlineMs: Long): Int
        @JvmStatic external fun nativeRelease()
    }
}
